app/scripts/context/veda-ui-provider.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { createContext, ReactNode, useContext } from 'react';
+
+interface EnvironmentConfig {
+  envMapboxToken: string;
+  envApiStacEndpoint: string;
+  envApiRasterEndpoint: string;
+}
+
+interface LinkProps {
+  to: string;
+  children: ReactNode;
+  [key: string]: any;
+}
+
+interface NavigationConfig {
+  LinkComponent: React.ElementType<Record<string, any>>;
+  linkPropName: string;
```

Comment:
(sorry this is not directly related to what this PR changes but since we are on it) - I wonder if we need to make this as an object (linkProps) and make the linkPropName as a part of it since we can't predict what props to expect.